content/shmem_team_destroy.tex

```diff
@@ -1,27 +1,27 @@
 \apisummary{
-    shmem\_team\_destroy function destroys existing team.
+    Destroys existing team.
 }
 
 \begin{apidefinition}
 
 \begin{Csynopsis}
-int @\FuncDecl{shmemx\_team\_destroy}@(shmem_team_t newteam);
+int @\FuncDecl{shmem\_team\_destroy}@(shmem_team_t team);
 \end{Csynopsis}
 
 \begin{apiarguments}
-\apiargument{IN}{newteam}{A valid SHMEM team handle.}
+\apiargument{IN}{team}{A valid SHMEM team handle.}
 \end{apiarguments}
 
 \apidescription{
-The shmemx\_team\_destroy function destroys an existing team. This is a
+The shmem\_team\_destroy function destroys an existing team. This is a
 collective call, in which every member of the team being destroyed needs
 to participate. This will free all internal memory structures associated
 with the team and invalidate the team handle. Upon return, the team
-handle is set to SHMEM\_TEAM\_NULL, after which it can no longer be
-used for team collective calls.
+handle is set to \CONST{SHMEM\_TEAM\_NULL}, after which it can no longer be
+used for team API calls.
 
-It is considered erroneous to free SHMEM\_TEAM\_WORLD or
-SHMEM\_TEAM\_NODE. Error checking will be done to ensure a valid
+It is considered erroneous to free \CONST{SHMEM\_TEAM\_WORLD} or
+\CONST{SHMEM\_TEAM\_NODE}. Error checking will be done to ensure a valid
 team handle is provided. All errors are considered fatal, and will
 result in the job aborting with an informative error message.
```

Comment:
As with #12, if an implementation is required to check the validity of the team handle, I think this call should return -1 when provided an invalid team handle instead of causing a fatal error. A value of -1 should be a sufficient sentinel, since no team may contain -1 PEs.

Reply:
Agreed in general on abort being too drastic for many of these calls. I am looking at the existing spec for functions that return integer values for success or error. The ones that I am seeing (thread_init, context_create) return 0 on success, nonzero otherwise. But, as you say, we have some team functions here that have 0 as a valid return value. So for team destroy, it seems better to have an error return value that matches the other team functions, i.e. -1 for failure.